queries.yaml

```diff
@@ -1421,12 +1421,64 @@ post_process:
       where: >-
         landuse_kind in ('residential', 'industrial')
 
-  # merge any remaining polygons by their properties.
+  # drop osm_relation tag, used to indicate that shapes came from OSM relations.
+  # however, it interferes with merging.
+  - fn: vectordatasource.transform.drop_properties
+    params:
+      source_layer: landuse
+      start_zoom: 4
+      end_zoom: 15
+      properties:
+        - osm_relation
+
+  # drop names on landuse areas at mid and low zooms. it interferes with
+  # merging, and anything that we would have labelled would have become a
+  # label by now anyway (see handle_label_placement around line 716-ish).
+  - fn: vectordatasource.transform.drop_names
+    params:
+      source_layer: landuse
+      start_zoom: 4
+      end_zoom: 13
+
+  # drop small inners before merge, as not having to consider these can speed up
+  # the buffer & clipping algorithms.
+  - fn: vectordatasource.transform.drop_small_inners
+    params:
+      end_zoom: 15
```

Comment:
Just curious, why is this 1 more zoom at 15 (versus 14) than in v1.6?

Reply:
I found that we still had some tiles at zoom 13 & 14 which had quite a lot of `grass` landuse. In the particular case I was looking at, they were all suburban lawns and, while the polygons couldn't be merged very well, they benefited from being merged into the same MultiPolygon.

So I added the [second `merge_polygon_features`](https://github.com/tilezen/vector-datasource/pull/1791/files/268f1e8d661ffa7d13b7245986c8f4804db4ab7d#diff-fc49e5c256610bc0b32a7de5a0520835R1468), which does a simpler merge step at zooms 13 & 14. I extended the range of the `drop_small_inners` to cover those zooms too, just to make sure we would be dropping any courtyards or cut-outs in those lawn features, or fields elsewhere.

For the lawns, I didn't get as substantial savings (only about 30% smaller for the layer - 15% smaller for the whole tile), and I figured we could just look at it when we have a global build and figure out whether this merging is doing what we want.